azkaban-common/src/main/c/execute-as-user.c

```diff
@@ -39,18 +39,23 @@ int INVALID_INPUT = 30;
 
 /*
  *  Change the real and effective user and group from super user to the specified user
- *  
+ *
  *  Adopted from:
  *  ./hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
- *  
+ *
  */
 
-int change_user(uid_t user, gid_t group) {
+int change_user(char *username, uid_t user, gid_t group) {
```

Comment:
I think `uid_t` is a typedef of an `unsigned int`. `char*` might be incorrect. This should be evident if you do a `*username` somewhere - might result in garbage or seg fault. Correct me if am wrong.